utils/node_local_test.py

```diff
@@ -1969,9 +1969,16 @@ class needs_dfuse_with_opt():
     # values are list of possible variants
     wrapped_names = {}
 
-    def __init__(self, caching=None, wbcache=True, dfuse_inval=True,
-                 ro=False):
-        self.caching = caching
+    # dict of tests that have been excluded at runtime (so we know what to skip)
+    # values are the versions of the test to skip
+    excluded_name_dict = {}
+
+    # ensure thread safety; not sure this is actually necessary, but meh...
+    wrapping_lock = threading.Lock()
```

Comment:
I think all dict operations are atomic. So if this code were changed to use a single `get`, it would be thread-safe
https://github.com/daos-stack/daos/blob/875b99cfebf6a56ab230c286932c7291ad91dbf0/utils/node_local_test.py#L2030-L2034

I.e.
```
     return list(needs_dfuse_with_opt.excluded_name_dict.get(name, [])
```